src/QMCDrivers/DMC/DMCBatched.cpp

```diff
@@ -232,19 +233,20 @@ void DMCBatched::advanceWalkers(const StateForThread& sft,
       std::transform(drifts.begin(), drifts.end(), log_gb.begin(),
                      [oneover2tau](auto& drift) { return -oneover2tau * dot(drift, drift); });
 
-      std::transform(ratios.begin(), ratios.end(), prob.begin(),
-                     [](auto ratio) { return std::norm(ratio); });
+      for (int iw = 0; iw < num_walkers; ++iw)
+      {
+        prob[iw] = std::norm(ratios[iw]) * std::exp(log_gb[iw] - log_gf[iw]);
+      }
 
       twf_accept_list.clear();
       twf_reject_list.clear();
       elec_accept_list.clear();
       elec_reject_list.clear();
 
-
       for (int iw = 0; iw < num_walkers; ++iw)
       {
         if ((!rejects[iw]) && prob[iw] >= std::numeric_limits<RealType>::epsilon() &&
-            step_context.get_random_gen()() < prob[iw] * std::exp(log_gb[iw] - log_gf[iw]))
+            step_context.get_random_gen()() < prob[iw])
```

Comment:
The above unchanged epsilon check gets affected by the change of prob.
This is actually an inconsistency between VMC and DMC implementations. The non-batched drivers also have it. It is hard to say which is right and which is wrong since the epsilon check is only a safeguard. I personally like the DMC way. So how about changing all the VMC to be consistent with DMC in a later PR?
After the change in this PR. The VMC non-batch and batch are consistent. The DMC non-batch and batch are consistent. VMC and DMC are not consistent.

Reply:
Well except the original DMC skips the epsilon comparison.  I don't think there is any point to that comparison so I think we should drop from both and make VMCBatched consistent.

Reply:
My original comment was incorrect and I updated it. @PDoakORNL was right there was check in the VMC driver only.
I still think this check is beneficial to kill bad sampling. Since random numbers are generated [0,1) there is still a chance that `0 < prob[iw]` gets valid. When prob gets very close to zero, the TWF is very singular and then all the numerics goes wrong. We get spikes in the energy and the Monte Carlo sampling cannot recover by itself. So I'd like to protect this case although the probability is low but it is possible.

Reply:
I will leave the check but since what we ultimately compare is the std::norm(ratio) * std::exp(log_gb[iw] - log_gf[iw])) I think it should be fine to compare that to epsilon.

Reply:
The check is intended to guard ratio close to zero namely the WF of the proposed move is very singular.